pyspider/libs/utils.py

```diff
@@ -106,30 +107,11 @@ def format_date(date, gmt_offset=0, relative=True, shorter=False, full_format=Fa
 
     format = None
     if not full_format:
-        if relative and days == 0:
-            if seconds < 50:
-                return ("1 second ago" if seconds <= 1 else
-                        "%(seconds)d seconds ago") % {"seconds": seconds}
-
-            if seconds < 50 * 60:
-                minutes = round(seconds / 60.0)
-                return ("1 minute ago" if minutes <= 1 else
-                        "%(minutes)d minutes ago") % {"minutes": minutes}
-
-            hours = round(seconds / (60.0 * 60))
-            return ("1 hour ago" if hours <= 1 else
-                    "%(hours)d hours ago") % {"hours": hours}
-
-        if days == 0:
-            format = "%(time)s"
-        elif days == 1 and local_date.day == local_yesterday.day and \
-                relative:
-            format = "yesterday" if shorter else "yesterday at %(time)s"
-        elif days < 5:
-            format = "%(weekday)s" if shorter else "%(weekday)s at %(time)s"
-        elif days < 334:  # 11mo, since confusing for same month last year
-            format = "%(month)s-%(day)s" if shorter else \
-                "%(month)s-%(day)s at %(time)s"
+        ret_, ret_format = fix_full_format(days, seconds, relative, shorter, local_date, local_yesterday)
```

Comment:
The refectory here is not correct. The original code is to generate a `format`, it has to be filled with actual date information in L140, it shouldn't be returned directly.

Reply:
I'm not sure what's incorrect compared to the original code. The original code has three "direct" return calls after the decision point "if relative and days == 0:" at line 109 of the original code. Which are the three cases we return in the refactoring.